config.nims

```diff
@@ -121,6 +121,34 @@ proc runUtil(f, util: string; args: seq[string]) =
   echo "Running '$1' .." % [cmd]
   exec cmd
 
+template preBuild(argos: string) =
+  assert argos.len > 0, "Build arguments must not be empty"
+  when defined(libressl) and defined(openssl):
+    error("Define only 'libressl' or 'openssl', not both.")
+  let (switches, nimFiles) = parseArgs()
+  assert nimFiles.len > 0, """
+    The 'musl' sub-command accepts at least one Nim file name
+      Examples: nim musl FILE.nim
+                nim musl FILE1.nim FILE2.nim
+                nim musl -d:pcre FILE.nim
+                nim musl -d:libressl FILE.nim
+                nim musl -d:pcre -d:openssl FILE.nim
+  """
+  var allBuildCmds {. inject .} = newSeqOfCap[array[0..1, string]](nimFiles.len)
+  for f in nimFiles:
+    let
+      extraSwitches = switches.mapconcat()
+      (dirName, baseName, _) = splitFile(f)
+      binFile = dirName / baseName  # Save the binary in the same dir as the nim file
+      nimArgsArray = when doOptimize:
+                       [argos, "-d:musl", "-d:release", "--opt:size", "--passL:-s", extraSwitches, " --out:" & binFile, f]
+                     else:
+                       [argos, "-d:musl", extraSwitches, " --out:" & binFile, f]
+      nimArgs = nimArgsArray.mapconcat()
+    allBuildCmds.add [nimArgs, binFile]
```

Comment:
I'd suggest making `allBuildCmds` a seq of named tuples with fields `nimArgs` and `binFile`. That way, later when you loop through it, you can do `cmd.nimArgs` and `cmd.binFile` (instead of using 0 and 1 indices right now).